iron-list.html

```diff
@@ -1244,6 +1244,15 @@
         var item = this.items && this.items[vidx];
         if (item != null) {
           inst[this.as] = item;
+
+          if(typeof item === 'object' && !Array.isArray(item)) {
+            for(var prop in item) {
+              if(item.hasOwnProperty(prop)) {
+                inst.set([this.as, prop], item[prop]);
```

Comment:
yup this has three problems:

1. it changes behaviour of iron-list a lot, in that elements didn't used to get `undefined` (or `null`) as their model but now will
2. it can trigger observers which weren't triggered before (at that point), because you set the value of the properties to non-undefined sooner
3. due to observers firing, if any trigger a `Polymer.dom.flush()`, our `_restoreFocusedItem` method throws null reference exceptions

Regarding 3, this is another issue. [This condition](https://github.com/PolymerElements/iron-list/blob/master/iron-list.html#L1768https://github.com/PolymerElements/iron-list/blob/master/iron-list.html#L1768) may pass (non-null offscreen element), but then if `assignModels` causes any observers to fire which trigger a flush, the offscreen element can become null so we mistakingly then set a physical item as null.